crucible-llvm/src/Lang/Crucible/LLVM/Translation/Instruction.hs

```diff
@@ -1847,6 +1866,30 @@ callFunctionWithCont instr tailCall_ fnTy fn args assign_f k
 
      | otherwise = callFunction (Just instr) tailCall_ fnTy fn args assign_f >> k
 
+-- | Match the arguments used by @dbg.addr@, @dbg.declare@, and @dbg.value@.
+dbgArgs :: 
+  [L.Typed L.Value] {- ^ debug call arguments -} ->
+  LLVMGenerator s arch ret (LLVMExpr s arch, L.DILocalVariable, L.DIExpression)
+dbgArgs args =
+  do (valArg, lvArg, diArg) <-
+       case args of
+         [valArg, lvArg, diArg] -> pure (valArg, lvArg, diArg)
+         _ -> fail "dbg expected 3 arguments"
```

Comment:
Similar request here - is it possible to include extra contextual information about what we got that was unexpected?